frontend/src/features/homes/add-group.vue

```diff
@@ -0,0 +1,73 @@
+<template>
+  <q-dialog
+    class="fit"
+    v-model="show"
+    @hide="$emit('close')"
+    @escape-key="$emit('close')"
+  >
+    <q-card>
+      <q-card-section>
+        <div class="text-h4">
+          <q-icon name="fa fa-cubes" class="q-mr-md" />
+          <span v-if="group">
+            {{ $i18n.t("groupModal-header-edit") }}
+          </span>
+          <span v-else>
+            {{ $i18n.t("groupModal-header-createNew") }}
+          </span>
+        </div>
+      </q-card-section>
+
+      <q-separator />
+      <q-card-section>
+        <q-form @submit="onSubmit" ref="manageGroup" class="q-gutter-md">
+          <q-input
+            v-model="groupName"
+            :rules="[
+              v =>
+                (!!v && v.length < 31) ||
+                $i18n.t('groupModal-group-requiredMaxLength')
```

Comment:
Please wrap this rule in a meaningful function name, like `maxLengthRule`. Inside the `maxLengthRule` function, use a meaningful variable name, like `maxLength` for the value of `31`. 

These types of habits make code much cleaner/easier to understand.